src/rgw/rgw_sync_module_es.cc

```diff
@@ -103,6 +103,55 @@ class ItemList {
 #define ES_NUM_SHARDS_DEFAULT 16
 #define ES_NUM_REPLICAS_DEFAULT 1
 
+struct ESVersion {
+  int major_ver;
+  int minor_ver;
+
+  ESVersion(int _major, int _minor): major_ver(_major), minor_ver(_minor) {}
+  ESVersion(): major_ver(0), minor_ver(0) {}
+
+  void from_str(const char* s) {
+    sscanf(s, "%d.%d", &major_ver, &minor_ver);
+  }
+
+  std::string to_str() const {
+    return std::to_string(major_ver) + "." + std::to_string(minor_ver);
+  }
+
+  void decode_json(JSONObj *obj);
+};
+
+bool operator >= (const ESVersion& v1, const ESVersion& v2)
+{
+  if (v1.major_ver == v2.major_ver)
+    return v1.minor_ver >= v2.minor_ver;
+
+  return v1.major_ver > v2.major_ver;
```

Comment:
shouldn't it be `>=` here?

Reply:
I guess the first equality should catch this, having said that this does look non intuitive, I'll collapse this to a one liner